src/main/java/com/moodybluez/enterprise/controller/EntryController.java

```diff
@@ -27,15 +24,17 @@ public Entry getById(@PathVariable int id) {
         List<Entry> entries = entryService.fetchByMonth(year, month);
         Map<Integer, Entry> ret = new HashMap<>();
         for(Entry entry: entries){
-            ret.put(entry.getDate().getDay(),entry);
+            Calendar calendarInstance = Calendar.getInstance();
+            calendarInstance.setTime(entry.getDate());
+            int dayOfWeek = calendarInstance.get(Calendar.DAY_OF_WEEK);
+            ret.put(dayOfWeek,entry);
```

Comment:
Technical Debt - The method being used here "getDay()" has been deprecated for a long time and this could really be a problem, especially if JVM is running newer version which will require you to update this anyway, thus creating technical debt.